### PR TITLE
feat(breaking): Drop support for Node 10; Require Node 12+

### LINKED ISSUE #9545: Remove support for Node.js 10 during `cypress install`
### Current behavior:

Cypress currently supports installation using Node.js 10.0.0+ 

[Node 10 end of life is April 30th, 2021](https://nodejs.org/en/about/releases/)

This should allow us to update some other dependencies:

- extract-zip: https://github.com/cypress-io/cypress/issues/6896

### Desired behavior:

We should remove support for Node.js 10 during `cypress install` and support Node.js 12.0.0 as a minimum. 

### Versions

Cypress 6.0.0

### PR DESCRIPTION
**Pull request by jennifer-shehane**
- close #9545  

#### Breaking Changes

- Installing Cypress on your system now requires Node.js 12+. 

#### Dependencies 

- Upgraded `arch ` from `2.1.2` to `2.2.0`.
- Upgraded `dayjs ` from `1.9.3` to `1.10.4`.
- Upgraded `eventemitter2 ` from `6.4.2` to `6.4.3`.
- Upgraded `execa ` from `4.0.2` to `5.0.0`.
- Upgraded `fs-extra ` from `9.0.1` to `9.1.0`.
- Upgraded `is-ci ` from `2.0.0` to `3.0.0`.
- Upgraded `is-installed-globally ` from `0.3.2` to `0.4.0`.
- Upgraded `lodash ` from `4.17.19` to `4.17.21`.
- Upgraded `pretty-bytes ` from `5.4.1` to `5.6.0`.
- Upgraded `supports-color ` from `7.2.0` to `8.1.1`.

### Additional Details

- I updated some devDependencies too.
- I'm not sure how to test this against the `test-npm-module-on-minimum-node-version` repo with the new way we do next dev version
- PR opened to update https://github.com/cypress-io/cypress-test-node-versions against this branch to no longer test against Node 10. https://github.com/cypress-io/cypress-test-node-versions/pull/54
- Docs PR https://github.com/cypress-io/cypress-documentation/pull/3614

### CODE REVIEW

File: cli/package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "size": "t=\"cypress-v0.0.0.tgz\"; yarn pack --filename \"${t}\"; wc -c \"${t}\"; tar tvf \"${t}\"; rm \"${t}\";",
     "test": "yarn test-unit",
     "test-debug": "node --inspect-brk $(yarn bin mocha)",
-    "test-dependencies": "dependency-check . --no-dev",
+    "test-dependencies": "dependency-check . --missing --no-dev --verbose",
```
Comment (jennifer-shehane) [inline cli/package.json:16]: `dependency-check` changed their default args in 4.0, so this should be equivalent to what it used to be.

File: packages/server/__snapshots__/1_es_modules_spec.js [hunk 2]
```diff
@@ -92,7 +92,7 @@ Module build failed (from [..]):
 SyntaxError: /foo/bar/.projects/e2e/lib/fail.js: Unexpected token (2:0)
 
   1 | export default {
-> 2 | 
+> 2 |
```
Comment (jennifer-shehane) [inline packages/server/__snapshots__/1_es_modules_spec.js:95]: I have no idea why this was necessary, but it kept failing without this snapshot update.

File: circle.yml [hunk 3]
```diff
@@ -1312,7 +1312,7 @@ jobs:
   test-npm-module-on-minimum-node-version:
     <<: *defaults
     docker:
-      - image: cypress/base:10.0.0
+      - image: cypress/base:12.0.0
```
Comment (flotwig) [inline circle.yml:1315]:
```suggestion
      - image: cypress/base:12.0.0-libgbm
```

The original base-12.0.0 image didn't include libgbm, this should fix the failing CI stage

**Review by flotwig — APPROVED** [review]

**Comment by jennifer-shehane** [comment]
These tests are failing on main 7.0 branch - merging in and ignoring.

**Pull request merged**